### PR TITLE
#6211 Focus document preview elements when clicking their respective pipeline header node

### LINKED ISSUE #6211: Slice 1: Focus document builder preview element when clicking on a Document Renderer header node
Implementation sketch here: https://www.notion.so/pixiebrix/DocBuild-Brick-Toggle-74069e3308dd4c8f8c028bbc0c965b1a?pvs=4#12af52de27544185af19256b0c51c00f

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Closes #6211

## Reviewer Tips
- All changes to files in `elementsPreview/` is just wrapping the components in `forwardRef`. The rest of those files are unchanged.

## Discussion
- In order to fix this papercut: https://pixiebrix.slack.com/archives/C023KL47XV4/p1691602628607189
    - I went with the approach to add an event listener to the Document Preview element
    - I tried all 3 approaches discussed in the thread, and found this approach to be the cleanest and least invasive

## Demo

- https://www.loom.com/share/e46cd6871d2b44bb92fbaa23b27f47f1

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

File: src/components/documentBuilder/preview/ElementPreview.tsx [hunk 1]
```diff
@@ -50,6 +53,50 @@ export type ElementPreviewProps = {
...
+      scrollIntoView();
+    }
+
+    return () => {
+      // Cleanup the event listener to avoid multiple listeners being added for the same event,
+      // which would cause scrollIntoView to be called multiple times with out-of-date refs when the event is fired.
+      window.removeEventListener(
+        `${SCROLL_TO_DOCUMENT_PREVIEW_ELEMENT_EVENT}-${elementName}`,
+        scrollIntoView
+      );
```
Comment (BLoe) [inline src/components/documentBuilder/preview/ElementPreview.tsx:93]: Just curious, why do you need this event listener? What is this doing?

File: src/components/documentBuilder/preview/elementsPreview/Container.tsx [hunk 2]
```diff
@@ -40,11 +40,13 @@ const Container: React.FunctionComponent<ContainerProps> = ({
   elementName,
   isHovered,
   isActive,
+  elementRef,
   ...restPreviewProps
 }) => (
   <Component
     {...restPreviewProps}
     className={cx(props.className, className, documentTreeStyles.container)}
+    ref={elementRef}
```
Comment (BLoe) [inline src/components/documentBuilder/preview/elementsPreview/Container.tsx:49]: Should this be passing `elementRef` here?

File: src/components/documentBuilder/preview/ElementPreview.tsx (hunk 1 above)
Comment (BLoe) [reply src/components/documentBuilder/preview/ElementPreview.tsx:93]: Oh I'm blind I didn't read the comment right there 🤦

Comment (mnholtz) [reply src/components/documentBuilder/preview/ElementPreview.tsx:93]: Does this answer your question? https://www.loom.com/share/a72f213384d74a15aa01d3281d42b4c2

File: src/components/documentBuilder/preview/elementsPreview/Container.tsx (hunk 2 above)
Comment (BLoe) [reply src/components/documentBuilder/preview/elementsPreview/Container.tsx:49]: Ah this `Component` is a bootstrap component, not a child preview component, so I think this is fine as-is 🤷‍♂️

**Review by BLoe — APPROVED** [review]

**Pull request merged**